### ISSUE TITLE
Weird tryon results after running CP-Viton-plus on colab -  am I doing something wrong!

### ISSUE THREAD

**Issue by TamerElsherif** [minar09/cp-vton-plus#49]
Hi Matiur, @minar09 
I tried to run the cp-vton-plus on Google colab but it gave weird result images in the folder "/cp-vton-plus/result/TOM/test/try-on" see below images. Is this the correct folder of the try on results?   if not, where is the correct folder to look for try-on look?
can you please help me . I am still beginner in this.

**_Result images:_**
![cp-viton-plus-result 000020_0](https://user-images.githubusercontent.com/42250020/104113859-90d94780-5306-11eb-89ba-4aec68f59e1e.jpg)
![cp-viton-plus-result 000010_0](https://user-images.githubusercontent.com/42250020/104113866-9b93dc80-5306-11eb-93b0-5b6b4f248493.jpg)

**_Sample Images of /result/GMM/test/overlayed_TPS :_**
![overlayed_TPS 0000010_0](https://user-images.githubusercontent.com/42250020/104114069-ea427600-5308-11eb-8b72-d078dd38eca4.jpg)

**_Sample Images of /result/GMM/test/result_dir :_**
![Result_dir 0000010_0](https://user-images.githubusercontent.com/42250020/104114065-e57dc200-5308-11eb-82bc-3440629de88c.jpg)

**_Sample Images of /result/GMM/test/warp-cloth :_**
![Warp-cloth 0000010_0](https://user-images.githubusercontent.com/42250020/104114063-e0207780-5308-11eb-845a-1a0e24e7c6a8.jpg)

Thanks,

**Comment by minar09** [comment]
Hi @TamerElsherif , yes, the folder seems to be the correct one for the try-on results.

However, as you can see, before running the try-on (TOM) network, you need to run the warping (GMM) network first, so that the cloth is matched to the target human body. After running GMM, copy the warp-cloth and mask results to the data directory, and then run the TOM network.

Please follow the guideline carefully for your testing. Good luck!

**Comment by TamerElsherif** [comment]
> Hi @TamerElsherif , yes, the folder seems to be the correct one for the try-on results.
> 
> However, as you can see, before running the try-on (TOM) network, you need to run the warping (GMM) network first, so that the cloth is matched to the target human body. After running GMM, copy the warp-cloth and mask results to the data directory, and then run the TOM network.
> 
> Please follow the guideline carefully for your testing. Good luck!

Thank you for your reply @minar09 . I did it already. 
I got these results after running GMM, copying the warp-cloth and mask results folders to data and then I ran the TOM

**Comment by minar09** [comment]
Yes, your testing process seems to be okay. Is the result still same as before?
![image](https://user-images.githubusercontent.com/3684134/104147330-2a8d1b80-5411-11eb-9405-5aa119e56359.png)
See here, cloth is not warped. You can check if the GMM pretrained model is correctly loaded or something.

**Comment by trilokpadhi** [comment]
Hi @minar09 
At first , I tested the model with the test images you had given , the results are really really good , appreciate you for the awesome work that you have done 
I am also trying on a custom data . I am also getting similar results like @TamerElsherif 
I have followed all the process as given by you to try on a custom data 
extract the image-parse by LIPJPPNET ----> run dataset_neck_skin_connection.py ----> run body_binary_masking.py 
and also generated the openpose keypoints with the COCO18 model .

During testing also , I first run the GMM model and the TOM model , but still I am not getting any reasonable result . Attached here images for your reference

Thanks 
![000048_1](https://user-images.githubusercontent.com/20143249/104153123-8ce11e80-5407-11eb-93a6-bd842e2c09ea.jpg)
![000048_1](https://user-images.githubusercontent.com/20143249/104153132-910d3c00-5407-11eb-8e83-3006a2212310.jpg)
![001](https://user-images.githubusercontent.com/20143249/104153142-95395980-5407-11eb-8950-89b6ba1d7d0e.jpg)
![001](https://user-images.githubusercontent.com/20143249/104153170-a4b8a280-5407-11eb-920d-1dec5ae7c412.png)
![001](https://user-images.githubusercontent.com/20143249/104153172-a97d5680-5407-11eb-9fb1-f253c7de83a8.png)
![001](https://user-images.githubusercontent.com/20143249/104153182-ada97400-5407-11eb-9250-04ec83485f6c.png)
![001](https://user-images.githubusercontent.com/20143249/104153199-bb5ef980-5407-11eb-8169-f2c45330111f.jpg)
final try-on result 
![001](https://user-images.githubusercontent.com/20143249/104153239-d92c5e80-5407-11eb-9772-a0d4ecf03b84.jpg)








Trilok

**Comment by minar09** [comment]
Hi @trilokpadhi , please check why the cloth is not warped from the GMM testing. As I mentioned above, you can check whether the GMM model is correctly loaded and stuff.  If it's loading correctly, can you please post here sample GMM warping results? Thanks.

**Comment by TamerElsherif** [comment]
Hi @minar09 ,

can you check the below result. I think the GMM is loaded because the GMM warped image is warped a little bit, is that correct?
![2021-01-17_23-24-29](https://user-images.githubusercontent.com/42250020/104856672-aa891900-591c-11eb-80aa-8e48e1362f2b.jpg)

what do you i can do to make it work? 

[Colab instance of the project](https://colab.research.google.com/drive/10nCk-WESMk-oDEkDIE2ATcP2AhLZ5Gqu?usp=sharing)

thanks again.

**Comment by thaithanhtuan** [comment]
Hi @TamerElsherif , can you debug by printing something to make sure that the GMM load the checkpoints correctly? It looks like it is not loaded.
Regards.

**Comment by TamerElsherif** [comment]
@thaithanhtuan , i found it wasn't able to load GMM file correctly.
thanks a lot.

**Comment by ISL270** [comment]
> @thaithanhtuan , i found it wasn't able to load GMM file correctly.
> thanks a lot.

what did you do to make it load the GMM file correctly?

**Comment by Muaz65** [comment]
Hi! can you please share the link of your custom image inference notebook?

**Comment by AnjaliSinghTW** [comment]
@thaithanhtuan @TamerElsherif I am having this same issue of getting colored images. Can you please help by stating the issue and its solution?
